### PR TITLE
Feature: Added option to set default column sizes

### LINKED ISSUE #9867: Feature: Add menu option to set current directories column sizes as the default
### What's the Problem?

Every folder needs to be adjusted regarding the column size and shown columns.
This end up in an non consistent view experience. Every folder needs to be individually set.

**Steps to fix**
- Add a menu option when right clicking the details layout headers "Set column widths as default. This option will set the current column sizes as the default. Users should still be able to override the widths.

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #9867 

**Comments**
- When resetting layout options, page needs to be swapped in order to see changes (I'm not sure about how to proceed to fix this).

**Validation**
How did you test these changes?
- [x] Built and ran the app

### CODE REVIEW

**Comment by ferrariofilippo** [comment]
Last commit is to avoid 0px but still visible columns.

**Comment by yaichenbaum** [comment]
> I set the default widths for columns at 200 (this is used when the user resets layout options). Is this ok?

What did we use before?

**Comment by yaichenbaum** [comment]
> When resetting layout options, page needs to be swapped in order to see changes (I'm not sure about how to proceed to fix this).

This can be left as it is for now but we should track it in an issue, I had the same issue with #10089

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs [hunk 1]
```diff
@@ -67,42 +65,27 @@ public FolderLayoutModes LayoutMode
...
         }
 
         public uint GetIconSize()
         {
             if (LayoutMode == FolderLayoutModes.DetailsView)
-            {
                 return Constants.Browser.DetailsLayoutBrowser.DetailsViewSize; // ListView thumbnail
-            }
+
             if (LayoutMode == FolderLayoutModes.ColumnView)
```
Comment (yaichenbaum) [inline src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: Would a switch case work here?

File: src/Files.Backend/Services/Settings/ILayoutSettingsService.cs [hunk 2]
```diff
@@ -30,7 +30,37 @@ public interface ILayoutSettingsService : IBaseSettingsService, INotifyPropertyC
...
+
+		/// <summary>
+		/// Gets or sets a value indicating date modified column's default width
+		/// </summary>
+		double DateModifiedColumnWidth { get; set; }
+
+		/// <summary>
+		/// Gets or sets a value indicating item type column's default width
+		/// </summary>
+		double ItemTypeColumnWidth { get; set; }
```
Comment (yaichenbaum) [inline src/Files.Backend/Services/Settings/ILayoutSettingsService.cs:51]: Is "Item" redundant? `TypeColumnWidth` might be better.

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs (hunk 1 above)
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: We can't since not all ifs use `==`.

**Comment by ferrariofilippo** [comment]
> What did we use before?

Actually we're already using it, I hadn't seen it

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs (hunk 1 above)
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: I mean, I'm not sure a 3-cases switch would lead to performance improvements

Comment (yaichenbaum) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: It helps with code readability but no need to worry if it doesn't work here

Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: It would look something like this:

**Comment by yaichenbaum** [comment]
It looks like settings the widths as the default resets the overrides. Steps to reproduce
1. Reset preferences
2. Navigate to a directory (eg. folder 1) and change the columns widths
3. Navigate to a different directory (eg. folder 2) and set the column widths as the default
4. Go back to the folder 1 and see that it's now using the column widths from folder 2

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs (hunk 1 above)
Comment (yaichenbaum) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:77]: I think that's cleaner but let's keep as is for now.

**Comment by ferrariofilippo** [comment]
I'll investigate further, but actually everything seems to work 

https://user-images.githubusercontent.com/102259289/193914994-9b3d7c1d-8d34-4246-878d-5782899b20bf.mp4

**Comment by ferrariofilippo** [comment]
I tried both recloning the repository and testing on a different device (Win10) but I'm still not able to reproduce this bug

**Comment by yaichenbaum** [comment]
I can't reproduce it either

**Review by yaichenbaum — APPROVED** [review]
LGTM

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs [hunk 3]
```diff
@@ -1,548 +1,535 @@
...
+			userSettingsService.PreferencesSettingsService.NameColumnWidth = columns.NameColumn.UserLengthPixels;
+			userSettingsService.PreferencesSettingsService.DateModifiedColumnWidth = columns.DateModifiedColumn.UserLengthPixels;
+			userSettingsService.PreferencesSettingsService.DateCreatedColumnWidth = columns.DateCreatedColumn.UserLengthPixels;
+			userSettingsService.PreferencesSettingsService.TypeColumnWidth = columns.ItemTypeColumn.UserLengthPixels;
+			userSettingsService.PreferencesSettingsService.SizeColumnWidth = columns.SizeColumn.UserLengthPixels;
+			userSettingsService.PreferencesSettingsService.TagColumnWidth = columns.TagColumn.UserLengthPixels;
+		}
+
+		public static void ResetColumnsWidth()
+		{
```
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: I thought that when the user resets the preferences, columns widths too should be reset

File: src/Files.App/ServicesImplementation/Settings/PreferencesSettingsService.cs [hunk 4]
```diff
@@ -189,6 +189,66 @@ public FolderLayoutModes DefaultLayoutMode
 			set => Set((long)value);
 		}
 
+		public double TagColumnWidth
+		{
+			get => Get(200d);
+			set
+			{
+				if (value != 0)
```
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ServicesImplementation/Settings/PreferencesSettingsService.cs]: I'll remove it as this is no more needed (I wrote that to fix a bug that involved recycle bin columns, but I fixed that another way)

File: src/Files.App/ViewModels/FolderSettingsViewModel.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: Resetting the database resets only individual preferences for folders. It doesn't reset other options (such as which columns to show)

[the remark this answers is not in the document]
Comment (yaichenbaum) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: Where is the code for resetting the individual preferences?

[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: Files.App.ViewModels.SettingsViewModels.PreferencesViewModel.ResetLayoutPreferences()

[the remark this answers is not in the document]
Comment (yaichenbaum) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: Doesn't `FolderSettingsViewModel.DbInstance.ResetAll();` take care of resetting the column sizes?

[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: `FolderSettingsViewModel.DbInstance.ResetAll();` clears the database, which remove individual preferences for directories.
Default column sizes remain the same that were set by the user

[the remark this answers is not in the document]
Comment (yaichenbaum) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: The default size should stay the same, the clear database option is only meant to remove overrides.

[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ViewModels/FolderSettingsViewModel.cs:525]: ok

**Review by yaichenbaum — APPROVED** [review]
Really nice improvement! LGTM

File: src/Files.App/ServicesImplementation/Settings/PreferencesSettingsService.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ferrariofilippo) [reply src/Files.App/ServicesImplementation/Settings/PreferencesSettingsService.cs]: @yair100 I found the reason I put there that code. I opened #10239 to fix it

**Pull request merged**